presto-main/src/main/java/com/facebook/presto/operator/ByteArrayUtils.java

```diff
@@ -140,8 +137,36 @@ public static int writeLongsWithNulls(byte[] target, int targetIndex, long[] val
         return targetIndex;
     }
 
+    public static void writeByte(byte[] target, int targetIndex, byte value)
+    {
+        unsafe.putByte(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
+    }
+
+    public static void writeShort(byte[] target, int targetIndex, short value)
+    {
+        unsafe.putShort(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
+    }
+
+    public static void writeInt(byte[] target, int targetIndex, int value)
+    {
+        unsafe.putInt(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
+    }
+
     public static void writeLong(byte[] target, int targetIndex, long value)
     {
         unsafe.putLong(target, (long) targetIndex + ARRAY_BYTE_BASE_OFFSET, value);
     }
+
+    public static int fill(byte[] target, int targetIndex, int length, byte value)
+    {
+        unsafe.setMemory(target, targetIndex + ARRAY_BYTE_BASE_OFFSET, length, value);
+        return targetIndex + length;
+    }
+
+    public static void writeLengthPrefixedString(SliceOutput output, String value)
```

Comment:
This method doesn't seem to belong here. I'd move it into `BlockEncodingBuffers`.